### DISCUSSION TITLE
SSH command timed out

### DISCUSSION THREAD

**Discussion by tope22** [apache/airflow#33621] [Q&A] [answered]
After upgrading my airflow environment from v2.0.2 to v2.5.1, I started running into the issue of 'SSH command timed out' which I never experienced while using v2.0.2. In my environment I have multiple dags with each dag having different tasks with different durations.

I tried setting cmd_timeout to 8000 but it times out because some of my task runs longer than others. It's difficult to set a value for the parameter cmd_timeout.

My question- Is there a way to set cmd_timeout to None? I tried setting it at the global level under the extras but I get this error-
If connection parameters need to be added to Extra, please make sure they are in the form of a single, valid JSON object.
The following Extra parameters were not added to the connection:
{"no_host_key_check": true,"cmd_timeout": None, "conn_timeout": 500}

 I tried setting it at the dag level but it fails with this error
raise AirflowException(f"SSH operator error: exit status = {exit_status}")
airflow.exceptions.AirflowException: SSH operator error: exit status = 2

My SSHOperator version is 3.6.0

**Comment by Taragolis** [comment, marked as answer]
> The following Extra parameters were not added to the connection:
{"no_host_key_check": true,"cmd_timeout": None, "conn_timeout": 500}

There is some minor issue, `None` is Python specific type, and `null` is equivalent in JSON. So try to use this one

```json
{"no_host_key_check": true,"cmd_timeout": null, "conn_timeout": 500}
```

**Reply by tope22** [reply]
Thank you. That helped
